.github/workflows/retry-functional-test.yaml

```diff
@@ -0,0 +1,577 @@
+    - cron: "0 1-23 * * *"
+
+env:
+  # Go version
+  GOVER: '1.21.7'
+  GOPROXY: https://proxy.golang.org
+
+  # gotestsum version - see: https://github.com/gotestyourself/gotestsum
+  GOTESTSUM_VER: 1.10.0
+
+  # Helm version
+  HELM_VER: 'v3.12.0'
+  # KinD cluster version
+  KIND_VER: 'v0.20.0'
+  # Dapr version
+  DAPR_VER: '1.12.0'
+  DAPR_DASHBOARD_VER: '0.14.0'
+  # Kubectl version
+  KUBECTL_VER: 'v1.25.0'
+  # Azure Keyvault CSI driver chart version
+  AZURE_KEYVAULT_CSI_DRIVER_VER: '1.4.2'
+  # Azure workload identity webhook chart version
+  AZURE_WORKLOAD_IDENTITY_WEBHOOK_VER: '1.1.0'
+  # Container registry for storing container images
+  CONTAINER_REGISTRY: ghcr.io/radius-project/dev
+  # Container registry for storing Bicep recipe artifacts
+  BICEP_RECIPE_REGISTRY: ghcr.io/radius-project/dev
+  # The radius functional test timeout
+  FUNCTIONALTEST_TIMEOUT: 60m
+  # The Azure Location to store test resources
+  AZURE_LOCATION: westus3
+  # The base directory for storing test logs
+  RADIUS_CONTAINER_LOG_BASE: dist/container_logs
+  # The Radius helm chart location.
+  RADIUS_CHART_LOCATION: deploy/Chart/
+  # The region for AWS resources
+  AWS_REGION: 'us-west-2'
+  # The AWS account ID
+  AWS_ACCOUNT_ID: '${{ secrets.FUNCTEST_AWS_ACCOUNT_ID }}'
+  # The current GitHub action link
+  ACTION_LINK: '${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}'
+  # Server where terraform test modules are deployed
+  TF_RECIPE_MODULE_SERVER_URL: "http://tf-module-server.radius-test-tf-module-server.svc.cluster.local"
+  # The functional test GitHub app id
+  FUNCTIONAL_TEST_APP_ID: 425843
+  # Private Git repository where terraform module for testing is stored.
+  TF_RECIPE_PRIVATE_GIT_SOURCE: "git::https://github.com/radius-project/terraform-private-modules//kubernetes-redis"
+
+jobs:
+  build:
+    name: Build Radius for test
+    runs-on: ubuntu-latest
+    env:
+      DE_IMAGE: 'ghcr.io/radius-project/deployment-engine'
+      DE_TAG: 'latest'
+    outputs:
+      REL_VERSION: ${{ steps.gen-id.outputs.REL_VERSION }}
+      UNIQUE_ID: ${{ steps.gen-id.outputs.UNIQUE_ID }}
+      PR_NUMBER: ${{ steps.gen-id.outputs.PR_NUMBER }}
+      CHECKOUT_REPO: ${{ steps.gen-id.outputs.CHECKOUT_REPO }}
+      CHECKOUT_REF: ${{ steps.gen-id.outputs.CHECKOUT_REF }}
+      RAD_CLI_ARTIFACT_NAME: ${{ steps.gen-id.outputs.RAD_CLI_ARTIFACT_NAME }}
+      DE_IMAGE: ${{ steps.gen-id.outputs.DE_IMAGE }}
+      DE_TAG: ${{ steps.gen-id.outputs.DE_TAG }}
+    steps:
+      - name: Get GitHub app token
+        uses: tibdex/github-app-token@v1
+        id: get_installation_token
+        with: 
+          app_id: ${{ env.FUNCTIONAL_TEST_APP_ID }}
+          private_key: ${{ secrets.FUNCTIONAL_TEST_APP_PRIVATE_KEY }}
+      - name: Set up checkout target
+        run: |
+          echo "CHECKOUT_REPO=youngbupark/radius" >> $GITHUB_ENV
+          echo "CHECKOUT_REF=refs/heads/youngp/enable-retry-test" >> $GITHUB_ENV
```

Comment:
This workflow runs the test by using my forked repo. I want to validate retry is working as expected with more aggressive test.